src/validation.rs

```diff
@@ -256,55 +257,82 @@ impl Validator {
-                    let inouts: Vec<&&VariableIndexEntry> = declared_parameters
-                        .iter()
-                        .filter(|p| VariableType::InOut == p.get_variable_type())
-                        .collect();
-                    // if the called pou has declared inouts, we need to make sure that these were passed to the pou call
-                    if !inouts.is_empty() {
-                        let mut passed_params_idx = Vec::new();
-                        if let Some(s) = parameters.as_ref() {
-                            match s {
-                                AstStatement::ExpressionList { expressions, .. } => {
-                                    for (i, e) in expressions.iter().enumerate() {
-                                        // safe index of passed parameter
-                                        if let Ok((idx, _)) =
-                                            get_implicit_call_parameter(e, &declared_parameters, i)
-                                        {
-                                            passed_params_idx.push(idx);
-                                        }
-                                    }
-                                }
-                                _ => {
-                                    // safe index of passed parameter
-                                    if let Ok((idx, _)) =
-                                        get_implicit_call_parameter(s, &declared_parameters, 0)
-                                    {
-                                        passed_params_idx.push(idx);
-                                    }
-                                }
-                            }
-                        }
-                        // check if all inouts were passed to the pou call
-                        inouts.into_iter().for_each(|p| {
-                            if !passed_params_idx.contains(&(p.get_location_in_parent() as usize)) {
-                                self.stmt_validator.diagnostics.push(
-                                    Diagnostic::missing_inout_parameter(
-                                        p.get_name(),
-                                        operator.get_location(),
-                                    ),
+                if let Some(pou) = context.find_pou(operator) {
+                    let declared_parameters = context.index.get_declared_parameters(pou.get_name());
+                    let passed_parameters = parameters
+                        .as_ref()
+                        .as_ref()
+                        .map(ast::flatten_expression_list)
+                        .unwrap_or_default();
+
+                    let mut passed_params_idx = Vec::new();
+                    let mut are_implicit_parameters = true;
+                    // validate parameters
+                    for (i, p) in passed_parameters.iter().enumerate() {
+                        if let Ok((index, right, is_implicit)) =
```

Comment:
can we rename index to "location_in_parent", i think we call it like that in the VariableIndexEntry